freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/instance/InstanceUpdater.java

```diff
@@ -0,0 +1,45 @@
+package com.sequenceiq.freeipa.service.stack.instance;
+
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceStatus;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+
+import javax.inject.Inject;
+
+@Component
+public class InstanceUpdater {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InstanceUpdater.class);
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    public Set<InstanceMetaData> updateStatuses(Stack stack, InstanceStatus instanceStatus) {
+        Set<InstanceMetaData> checkableInstances = stack.getAllInstanceMetaDataList().stream()
```

Comment:
`Stack` already has a `getNotDeletedInstanceMetaDataSet` method, you could use that one

Reply:
Thanks! I wasn't aware of this existing function. Now I reuse it for this purpose.